src/compute-client/src/controller/instance.rs

```diff
@@ -997,29 +997,26 @@ where
         let mut storage_dependencies = Vec::new();
         let mut compute_dependencies = Vec::new();
 
+        // Any potentially acquired STORAGE read holds. We acquire them and
+        // check whether our as_of is valid. They are dropped once we installed
+        // read capabilities manually.
```

Comment:
Can you add a TODO here to store the read holds instead of installing read capabilities with the storage controller manually?